### ISSUE TITLE
setting `root` parameter of S3 filesystem produces incorrect paths for some tenant-aware jobs.

### ISSUE THREAD

**Issue by plakhin** [stancl/tenancy#739]
#### Describe the bug
When I set `root` parameter of S3 filesystem, some "tenant-aware" jobs get incorrect path when queues are being processed by Laravel Horizon. It is hard to "catch" the error, however I'll try to provide all steps to reproduce. The strangest thing for me is that if I dispatch a bunch of jobs for 1 tenant - all works good. If I dispatch jobs for 2 tenants - 2nd tenant gets wrong paths. But if I dispatch jobs for 2 tenants and restart Horizon after all jobs for 1st tenant were processed but before jobs for 2nd tenant start to process - all works fine again for 2nd tenant. If I don't set `root` - all works fine for all tenants.

#### Steps to reproduce
filesystems.php:
```
        's3' => [
            'driver'                  => 's3',
            'key'                     => env('AWS_ACCESS_KEY_ID'),
            'secret'                  => env('AWS_SECRET_ACCESS_KEY'),
            'region'                  => env('AWS_DEFAULT_REGION'),
            'bucket'                  => env('AWS_BUCKET'),
            'url'                     => env('AWS_URL'),
            'endpoint'                => env('AWS_ENDPOINT'),
            'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
            'visibility'              => 'public',
            'root'                    => 'media',
```
1. Create a simple queued job that will put a file to the `s3` disk: `Storage::cloud()->put('hello.json', '{"hello": "world"}');`
2. Dispatch that job for 2-3 tenants
3. Start horizon and wait till all jobs processed.
#### Expected behavior
There should be 3 files in S3 storage bucket:
```
/media/tenant1/hello.json
/media/tenant2/hello.json
/media/tenant3/hello.json
```

#### Actual behavior
There will be an error if using MinIO because the path containing one extra `/` symbol for Tenant2 (for Tenant 1 the path is correct).
The path for `put` command should be `/media/tenant2/hello.json` but the actual path is `/media//tenant2/hello.json`

There will be 3 files in S3 storage bucket (in case of Digital Ocean Spaces):
```
/media/tenant1/hello.json
/tenant2/hello.json
/tenant3/hello.json
```

#### Your setup
 - Laravel version: [e.g. 8.64.0]
 - stancl/tenancy version: [e.g. 3.4.4]

**Comment by stancl** [comment]
Hey, have you managed to find a solution to this? And is it an issue with the core of the package?

**Comment by plakhin** [comment]
Seems like the core issue. I didn't find a solution :( just not setting `root`.

**Comment by stancl** [comment]
Okay, will check this in detail during v4 development

**Comment by stancl** [comment]
Just to confirm — still didn't find any other cause? Planning v4 todos now.

**Comment by plakhin** [comment]
I didn't spend much time trying to find the solution, just sticked to settings without root.

**Comment by jtomek** [comment]
Hey, I just migrated to v9. The double forward slash was causing issues on S3. The generated URLs were https://bucket.s3.region.amazonaws.com//{tenant_prefix_and_id}/path which created a new folder "/" with tenant folders in it.

I have hot-fixed by implementing a custom FilesystemBootstrapper and replaced the code that prefixes the suffix from this:


```php
if (! $finalPrefix) {
    $finalPrefix = $originalRoot . '/'. $suffix;
}
```

to this:

```php

if (! $finalPrefix) {
    if (filled($originalRoot)) {
        $finalPrefix = $originalRoot . '/'. $suffix;
    } else {
        $finalPrefix = $suffix;
    }
}

```

I have not tested it yet fully and hope it won't break anything. 😵‍💫

**Comment by stancl** [comment]
Fixed in #828
